fairseq/checkpoint_utils.py

```diff
@@ -203,7 +203,7 @@ def load_checkpoint(cfg: CheckpointConfig, trainer, **passthrough_args):
         cfg.restore_file == "checkpoint_last.pt"
     ):  # default value of restore_file is 'checkpoint_last.pt'
         checkpoint_path = os.path.join(
-            cfg.save_dir, "checkpoint_last{}.pt".format(suffix)
+            cfg.get("save_dir"), "checkpoint_last{}.pt".format(suffix)
```

Comment:
That's a weird one:
save_dir is not in the config during the test.
In Hydra 1.0, this returned None.
In Hydra 1.1, this raises an error.

os.path.join normally would fail if it gets None, but it passes because the test is mocking it.
The fix here is just retaining the existing weird behavior (join with a None save_dir).